File: backend/src/main/java/fr/formation/backend/model/TypeIst.java
package fr.formation.backend.model;

public enum TypeIst {
    bacterie("Bactérienne"), virus("Virale"), parasite("Parasitaire"),
    bacterien("Bactérien"), viral("Viral");

    private String typeIst;

    private TypeIst(String typeIst) {
        this.typeIst = typeIst;
    }

    public String getTypeIst() {
        return typeIst;
    }

    public void setTypeIst(String typeIst) {
        this.typeIst = typeIst;
    }
}

File: frontend/src/app/enum/type-ist.ts
export type TypeIst = 'bacterienne' | 'virale' | 'parasite' | 'bacterien' | 'viral';

export const TypeIstEmoji: Record<TypeIst, { emoji: string; label: string }> = {
  bacterienne: { emoji: '🦠', label: 'Bactérienne' },
  virale: { emoji: '🫟', label: 'Virale' },
  parasite: { emoji: '🪱', label: 'Parasitaire' },
  bacterien: { emoji: '🦠', label: 'Bactérien' },
  viral: { emoji: '🫟', label: 'Viral' },
};
